### PR TITLE
Definitions for eventbusjs

### PR DESCRIPTION
**Pull request by Longqin88888**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/eventbusjs/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Type definitions for eventbusjs 0.2.0
+// Project: https://github.com/krasimir/EventBus
+
+export = EventBus;
+export as namespace EventBus;
+
+declare namespace EventBus {
+    function addEventListener(type: string, callback?: Function, scope?: Object): void;
```
Comment (codershiba) [inline types/eventbusjs/index.d.ts:8]:
```suggestion
    function addEventListener(type: string, callback?: Function, scope?: object): void;
```

File: types/eventbusjs/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    function removeEventListener(type: string, callback?: Function, scope?: Object): void;
```
Comment (codershiba) [inline types/eventbusjs/index.d.ts:10]:
```suggestion
    function removeEventListener(type: string, callback?: Function, scope?: object): void;
```

File: types/eventbusjs/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    function hasEventListener(type: string, callback?: Function, scope?: Object): boolean;
+
+    function dispatch(type: string, target?: Object, arg?: string, arg2?: string): void;
```
Comment (codershiba) [inline types/eventbusjs/index.d.ts:14]:
```suggestion
    function dispatch(type: string, target?: object, arg?: string, arg2?: string): void;
```

File: types/eventbusjs/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+export = EventBus;
+export as namespace EventBus;
+
+declare namespace EventBus {
+    function addEventListener(type: string, callback?: Function, scope?: Object): void;
+
+    function removeEventListener(type: string, callback?: Function, scope?: Object): void;
+
+    function hasEventListener(type: string, callback?: Function, scope?: Object): boolean;
```
Comment (codershiba) [inline types/eventbusjs/index.d.ts:12]:
```suggestion
    function hasEventListener(type: string, callback?: Function, scope?: object): boolean;
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/eventbusjs/package.json
1. types/eventbusjs/index.d.ts
1. types/eventbusjs/eventbusjs-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b8602cd1d15866edd4c869c7731b6d76f5abaff8
</p>

File: types/eventbusjs/index.d.ts (hunk 1 above)
Comment (Longqin88888) [reply types/eventbusjs/index.d.ts]: Completed

File: types/eventbusjs/index.d.ts (hunk 2 above)
Comment (Longqin88888) [reply types/eventbusjs/index.d.ts]: Completed

File: types/eventbusjs/index.d.ts (hunk 3 above)
Comment (Longqin88888) [reply types/eventbusjs/index.d.ts]: Completed

File: types/eventbusjs/index.d.ts (hunk 4 above)
Comment (Longqin88888) [reply types/eventbusjs/index.d.ts]: Completed

**Review by ghost — APPROVED** [review]

**Review by sheetalkamat — APPROVED** [review]

**Comment by Longqin88888** [comment]
Ready to merge

**Pull request merged**